src/dappsExplorer/DAppsExplorerScreen.tsx

```diff
@@ -0,0 +1,352 @@
+      />
+      {ConfirmOpenDappBottomSheet}
+      <>
+        {!loading && error && (
+          <View style={styles.centerContainer}>
+            <Text style={fontStyles.regular}>{t('dappsScreen.errorMessage')}</Text>
+          </View>
+        )}
+        {categories.length && (
+          <AnimatedSectionList
+            refreshControl={
+              <RefreshControl
+                tintColor={colors.greenBrand}
+                colors={[colors.greenBrand]}
+                style={{ backgroundColor: colors.light }}
+                refreshing={loading}
+                onRefresh={() => dispatch(fetchDappsList())}
+              />
+            }
+            // @ts-ignore TODO: resolve type error
+            ref={sectionListRef}
+            ListFooterComponent={
+              <>
+                {dappsMinimalDisclaimerEnabled ? (
+                  <Text style={styles.disclaimer}>{t('dappsDisclaimerAllDapps')}</Text>
+                ) : null}
+              </>
+            }
+            ListHeaderComponent={
+              <>
+                <DescriptionView
+                  title={t('dappsScreen.title')}
+                  message={t('dappsScreen.message')}
+                />
+                {/* Dapps Filtering*/}
+                <View style={{ paddingTop: Spacing.Thick24 }}>
+                  <ScrollView
+                    horizontal={true}
+                    // Expand the scrollview to the edges of the screen
+                    style={{ marginHorizontal: -Spacing.Thick24 }}
+                    showsHorizontalScrollIndicator={false}
+                    bounces={false}
+                    ref={horizontalScrollView}
+                  >
+                    {/* All Dapps Filter */}
+                    <DappFilterChip
+                      chipFilter={{ id: 'all', name: t('dappsScreen.allDapps') }}
+                      selectedFilter={selectedFilter}
+                      setFilter={setSelectedFilter}
+                      lastChip={false}
+                      key={'all'}
+                    />
+                    {/* Category Filter Chips */}
+                    {categories.map((category, idx) => {
+                      return (
+                        <DappFilterChip
+                          chipFilter={{ id: category.id, name: category.name }}
+                          selectedFilter={selectedFilter}
+                          setFilter={setSelectedFilter}
+                          lastChip={idx === categories.length - 1}
+                          key={category.id}
+                        />
+                      )
+                    })}
+                  </ScrollView>
+                </View>
+                {dappFavoritesEnabled && (
+                  <>
+                    <Text style={styles.sectionTitle}>{t('dappsScreen.favoriteDappsUpper')}</Text>
+                    <FavoriteDappsSection
+                      filter={selectedFilter}
+                      removeFilter={() => {
```

Comment:
the remove filter functionality is repeated, can we abstract this into a function?

Reply:
Fixed in 55b20eb68d73ebf49213e9c26a4ae5f7a8d34e89!